### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by jdogmcsteezy**

### CODE REVIEW

**Review by dwelch-spike — COMMENTED** [review]
Just a few comments. Also linting is failing

File: docker/auth/docker-compose.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ services:
       timeout: 20s
       retries: 20
   avs:
-    image: aerospike.jfrog.io/docker/aerospike/aerospike-vector-search-private:0.9.1-SNAPSHOT
+    image: aerospike/aerospike-vector-search:0.10.0
```
Comment (reugn) [inline docker/auth/docker-compose.yml:19]: There is an option to set the image versions in the docker compose files using [interpolation](https://docs.docker.com/compose/how-tos/environment-variables/variable-interpolation/).

File: connection_provider.go [hunk 2]
```diff
@@ -566,40 +606,40 @@ func (cp *channelProvider) checkAndSetNodeConns(
 
 // removeDownNodes removes the gRPC client connections for nodes in nodeConns
 // that aren't apart of newNodeEndpoints
```
Comment (reugn) [inline connection_provider.go:608]: a part

File: makefile [hunk 3]
```diff
@@ -1,34 +1,70 @@
...
+	$(MOCKGEN) --source protos/vector-db_grpc.pb.go --destination protos/vector-db_grpc_mock.pb.go --package protos
+
+
 .PHONY: test
-test: integration unit
+test: unit integration 
 
 .PHONY: integration
-integration:
+integration: $(GOLEAK)
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply makefile:48]: not needed

File: testutils.go [hunk 4]
```diff
@@ -79,11 +82,24 @@ func GetUint32Ptr(i int) *uint32 {
...
 }
 
+func GetBoolPtr(b bool) *bool {
+	return &b
+}
+
+func GetTimePtr(t time.Time) *time.Time {
+	return &t
+}
+
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply testutils.go:102]: cool, thanks

File: token_manager.go [hunk 5]
```diff
@@ -74,6 +73,11 @@ func (tm *tokenManager) RefreshToken(ctx context.Context, conn grpc.ClientConnIn
 	}
 
 	claims := strings.Split(resp.GetToken(), ".")
+
+	if len(claims) < 3 {
+		return fmt.Errorf("failed to authenticate: missing either header, payload, or signature")
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply token_manager.go:78]: I was under the assumption that we should avoid logging auth info. Is that not the case this time?

[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply token_manager.go:78]: Maybe since it is an error case it is ok?

File: docker/auth/docker-compose.yml (hunk 1 above)
Comment (jdogmcsteezy) [reply docker/auth/docker-compose.yml:19]: thanks

added a docker/.env file

File: connection_provider.go [hunk 6]
```diff
@@ -50,51 +64,65 @@ func newConnection(conn *grpc.ClientConn) *connection {
...
+	nodeConnsLock   *sync.RWMutex
+	tendInterval    time.Duration
+	clusterID       uint64
+	listenerName    *string
+	isLoadBalancer  bool
+	token           tokenManager
+	stopTendChan    chan struct{}
+	closed          atomic.Bool
+	grpcConnFactory func(hostPort *HostPort) (grpcClientConn, error) // For testing
+	connFactory     func(conn grpcClientConn) *connection            // For testing
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply connection_provider.go]: still thinking about how to best clean this up.

File: token_manager.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (reugn) [reply token_manager.go:78]: I missed out that it is the authentication token. You are right.

**Review by reugn — APPROVED** [review]

**Pull request merged**